Maps/ArchipelagoCampaign/WoL/ap_engine_of_destruction.SC2Map/MapScript.galaxy

```diff
@@ -548,6 +549,76 @@ void gt_ArchInit_Init () {
     TriggerAddEventMapInit(gt_ArchInit);
 }
 
+//--------------------------------------------------------------------------------------------------
+// Trigger: FactionSwapInit
+//--------------------------------------------------------------------------------------------------
+bool gt_FactionSwapInit_Func (bool testConds, bool runActions) {
+    // Automatic Variable Declarations
+    unitgroup autoAFAA1727_g;
+    int autoAFAA1727_u;
+    unit autoAFAA1727_var;
+    unitgroup autoB4A30A31_g;
+    int autoB4A30A31_u;
+    unit autoB4A30A31_var;
+
+    // Actions
+    if (!runActions) {
+        return true;
+    }
+
+    if ((libABFE498B_gv_aP_Triggers_Option_overridePlayerRace == "Neut")) {
+        lib15EF4C78_gf_RemoveUnusedStartingUnits(UnitGroupFromId(5), "Terr", false);
+    }
+    else {
+        libABFE498B_gv_aP_Triggers_skipCutscenes = true;
+        lib15EF4C78_gf_RemoveUnusedStartingUnits(UnitGroupFromId(5), libABFE498B_gv_aP_Triggers_Option_overridePlayerRace, false);
+        libLbty_gf_OrderWorkerstoGatherNearbyResources(RegionEntireMap(), 1);
+        lib15EF4C78_gf_AP_Player_UtilTownHallAutoRally(1);
+        if ((libABFE498B_gv_aP_Triggers_Option_overridePlayerRace == "Zerg")) {
+            PlayerSetRace(1, "Zerg");
+            lib5BD4895D_gf_AP_Core_MapConfig_setPlayerFaction(1, lib5BD4895D_gv_aP_Core_Faction_KERRIGAN_SWARM);
+            autoB4A30A31_g = UnitGroup(null, gv_p01_USER, RegionFromId(3), UnitFilter(0, 0, (1 << c_targetFilterMissile), (1 << (c_targetFilterDead - 32))), 0);
+            autoB4A30A31_u = UnitGroupCount(autoB4A30A31_g, c_unitCountAll);
+            for (;; autoB4A30A31_u -= 1) {
+                autoB4A30A31_var = UnitGroupUnitFromEnd(autoB4A30A31_g, autoB4A30A31_u);
+                if (autoB4A30A31_var == null) { break; }
+                libNtve_gf_PauseUnit(autoB4A30A31_var, true);
+                libNtve_gf_ShowHideUnit(autoB4A30A31_var, false);
+            }
+        }
+        else {
+            if ((libABFE498B_gv_aP_Triggers_Option_overridePlayerRace == "Prot")) {
+                PlayerSetRace(1, "Prot");
+                lib5BD4895D_gf_AP_Core_MapConfig_setPlayerFaction(1, lib5BD4895D_gv_aP_Core_Faction_DAELAAM);
+                if ((libABFE498B_gv_aP_Triggers_Option_SoAPresence == 2)) {
+                    lib15EF4C78_gf_AP_Player_CreateSoAUnit(1);
+                }
+
+                if ((libABFE498B_gv_aP_Triggers_Option_SoAPresenceAutocasts == 2)) {
+                    lib15EF4C78_gf_AP_Player_CreateSoAAutonomousCasterUnit(1);

```

Comment:
I prefer using constants for players, each map has those

Reply:
I double checked the code and I don't think this is an issue? It uses the same triggers as all of the other raceswap maps.